src/iexamine.cpp

```diff
@@ -5927,7 +5927,7 @@ static void smoker_finalize( Character &, const tripoint &examp, const time_poin
             } else {
                 it.calc_rot_while_processing( 6_hours );
 
-                item result( it.get_comestible()->smoking_result, start_time + 6_hours, it.charges );
+                item result( it.get_comestible()->smoking_result, start_time + 6_hours, it.count() );
```

Comment:
```suggestion
                item result( it.get_comestible()->smoking_result, start_time + 6_hours );
                if( it.count_by_charges() ) {
                    result.charges = it.charges;
                }
```

To avoid setting charges on non-charge items. It shouldn't cause issues if it's only set to 1, but it's cleaner nonetheless.

Reply:
I see it in annotation, method `count()` already includes two conditions that you are concerned . Still I should accept your suggestion in case.

Reply:
Tested S/L during smoking again, with no new problem.

Reply:
> I see it in annotation, method `count()` already includes two conditions that you are concerned . Still I should accept your suggestion in case.

The difference is having 0 or 1 charges on an item that isn't allowed to have charges. The latter will be migrated and set to 0 when you load a save, but as long as it's just 1 charge nothing really happens. But it's still not supposed to happen.